src/rust/ide/view/graph-editor/src/component/visualization/container.rs

```diff
@@ -608,6 +615,20 @@ impl Container {
             eval_ on_selected ( action_bar.hide_icons.emit(()) );
         }
 
+
+        // ===  Action bar actions ===
+        frp::extend! { network
+
+            eval_ action_bar.action_reset_position([model]{
+               model.drag_root.set_position_xy(Vector2::zero());
+            });
+
+            previous_cursor <- app.cursor.frp.scene_position.previous();
+            position_change <- app.cursor.frp.scene_position.map2(&previous_cursor, |p1,p2| p2.xy() - p1.xy());
+            drag_action <- position_change.gate(&action_bar.action_drag_container);
+            eval drag_action ([model](mouse) model.drag_root.mod_position_xy(|pos| pos - *mouse));
```

Comment:
please simplify it (the same way as requested in other FRP comments)

Reply:
I added the `scene_position` so I'm pretty sure it is not in there. But you are right, this logic probably should be exposed from there anyway. So, that is done now.